ape_hardhat/provider.py

```diff
@@ -1001,45 +1001,51 @@ def get_virtual_machine_error(self, exception: Exception, **kwargs) -> VirtualMa
         elif message.startswith("execution reverted: "):
             message = message.replace("execution reverted: ", "")
 
+        def _handle_execution_reverted(revert_message: Optional[str] = None, **kwargs):
+            if revert_message in ("", "0x", None):
+                revert_message = TransactionError.DEFAULT_MESSAGE
+
+            enriched = self.compiler_manager.enrich_error(
+                ContractLogicError(revert_message=revert_message, **kwargs)
+            )
+
+            if enriched.message == TransactionError.DEFAULT_MESSAGE and revert_message:
+                # Since input data is always missing, and to preserve backwards compat,
+                # use the selector as the message still.
+                enriched.message = revert_message
+
+            # Show call trace if availble
+            if enriched.txn:
+                # Apparently this is not always a receipt on a failed tx
```

Comment:
Correct, but likely if that is the case, tx.receipt will be None. For example, if your tx fails during estimation, there won't be a receipt.

Reply:
It does check if the receipt exists before attempting to show the trace.  Mostly just covering all the possibilities here.